spec/features/post_index_page_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe 'User', type: :feature do
+  before :each do
+    first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
+    second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+                              bio: 'Teacher from Poland.')
+    first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
+    Comment.create(post: first_post, author: second_user, text: 'Hi Tom!')
+    Comment.create(post: first_post, author: second_user, text: 'Hello Tom!')
+    visit user_posts_path(first_user)
+  end
+
+  describe 'index page' do
+    it "I can see the user's profile picture" do
+      all_images = page.all('img')
+      all_images.each do |img|
+        link = img[:src]
+        expect(link).to eq('/assets/img2-3b2280af3d478258894911b061e6c8002e3a687c7ad0568da200cb39dc5e35f4.jpg')
+      end
+    end
+
+    it "I can see the user's username" do
+      expect(page).to have_content('Tom')
+      expect(page).to have_content('Lilly')
+    end
+
+    it 'I can see the number of posts the user has written' do
+      expect(page).to have_content('Number of posts: 1')
+    end
+
+    it "I can see a post's title" do
+      expect(page).to have_content('Hello')
+    end
+
+    it 'I can see the first 3 comments on a post' do
+      # Add code to create and display first 3 posts
+      expect(page).to have_content('Hi Tom!')
+    end
```

Comment:
- [ ] I think it would be better if you update this test so it has tests that asserts you can see the first 3 comments on a post as required for the project